flutter/news/lib/features/frontpage/data/repositories/articles_repository.dart

```diff
@@ -1,43 +1,26 @@
-import 'package:dartz/dartz.dart';
-import 'package:news/core/error/exceptions.dart';
+import 'package:news/core/result.dart';
 
-import '../../../../core/error/failures.dart';
-import '../../../../core/network/network_info.dart';
 import '../../domain/entities/article.dart';
 import '../datasource/articles_data_source.dart';
 import '../datasource/articles_local_data_source.dart';
 
 class ArticlesRepository {
   final ArticlesLocalDataSource localDataSource;
   final ArticlesRemoteDataSource remoteDataSource;
-  final NetworkInfo networkInfo;
 
   ArticlesRepository(
-      {required this.localDataSource,
-      required this.remoteDataSource,
-      required this.networkInfo});
+      {required this.localDataSource, required this.remoteDataSource});
 
-  Future<Either<Failure, List<Article>>> getEverythingAbout(String query) {
+  Future<Result<List<Article>>> everythingAbout(String query) {
     // TODO: implement getEverythingAbout
     throw UnimplementedError();
   }
 
-  Future<Either<Failure, List<Article>>> getTopHeadlines() async {
-    if (await networkInfo.isConnected) {
-      try {
-        final remoteArticles = await remoteDataSource.getTopHeadlines();
-        localDataSource.cacheTopHeadlines(remoteArticles);
-        return Right(remoteArticles);
-      } on ServerException catch (e) {
-        return Left(ServerFailure(e.message));
-      }
-    } else {
-      try {
-        final localArticles = await localDataSource.getLastTopHeadlines();
-        return Right(localArticles);
-      } on CacheException catch (e) {
-        return Left(CacheFailure(e.message));
-      }
+  Future<Result<List<Article>>> topHeadlines() async {
+    final result = await remoteDataSource.topHeadLines();
```

Comment:
As explained on the description I am aiming for simplifying this logic.

The idea is to use the `LocalDataSource` as single source of truth. So that we:

1. Try to fetch from remote
2. Save result on remote , if succeeded
3. Regardless of what happened, retrieves the last thing that was saved on local

This simplifies testing and also makes the logic simpler to follow